rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +394,67 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED and \
+      participant_summary.samplesToIsolateDNA == SampleStatus.RECEIVED:
```

Comment:
Similar logic with ordered time, just add one more criteria samplesToIsolateDNA

Reply:
It'd be nice if we could have a helper function that both calculate_core_stored_sample_time and calculate_core_ordered_sample_time call, with the shared logic.

Reply:
add a function for the dup logic